app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -232,7 +232,7 @@ def action_permission
     end
 
     def find_content_overrides
-      if params[:content_overrides_search]
+      if params.dig(:content_overrides_search, :search).present?
         content_labels = ::Katello::Content.joins(:product_contents)
                             .where("#{Katello::ProductContent.table_name}.product_id": @host.organization.products.subscribable.enabled)
                             .search_for(params[:content_overrides_search][:search])
```

Comment:
When calling hammer to enable/disable a repo, `params` has `"content_overrides_search" => {}`.
When calling hammer to reset a repo, `params` has `"content_overrides_search" => {"remove"=>true}`.

For above two cases, `params[:content_overrides_search]` exists but `params[:content_overrides_search][:search]` is `nil` and this search returns back all repos. That is why all repos were affected by the hammer commands.